### PR TITLE
refactor(archive,async,cli,csv,dotenv,encoding,expect,fmt,front-matter,fs,http,internal,log,net,path,semver,testing,text,webgpu,yaml): enable `"exactOptionalPropertyTypes"` option

### PR DESCRIPTION
**Pull request by petamoriken**
Related https://github.com/denoland/deno/issues/25162

By including this PR, users will be able to use std modules without problems when the `“exactOptionalPropertyTypes”` option (which is recommended in [tsconfig reference](https://www.typescriptlang.org/tsconfig/#exactOptionalPropertyTypes)) is enabled.

### CODE REVIEW

**Review by iuioiua — COMMENTED** [review]
I'm for strictness. Though, seeing as tests have to be modified, this seems like a breaking change. WDYT, @kt3k?

File: csv/parse_stream_test.ts [hunk 1]
```diff
@@ -510,7 +511,9 @@ Deno.test({
 
     // skipFirstRow option
     {
-      const { readable } = new CsvParseStream({ skipFirstRow: undefined });
+      const { readable } = new CsvParseStream(
+        { skipFirstRow: undefined } as object,
```
Comment (kt3k) [inline csv/parse_stream_test.ts:515]: Can this be `{}` instead?

File: csv/parse_test.ts [hunk 2]
```diff
@@ -960,7 +960,7 @@ Deno.test({
 
     // skipFirstRow option
     {
-      const parsed = parse("a\nb", { skipFirstRow: undefined });
+      const parsed = parse("a\nb", { skipFirstRow: undefined } as object);
```
Comment (kt3k) [inline csv/parse_test.ts:963]:
```suggestion
      const parsed = parse("a\nb", {});
```

File: csv/parse_test.ts [hunk 3]
```diff
@@ -976,7 +976,7 @@ Deno.test({
 
     // columns option
     {
-      const parsed = parse("a\nb", { columns: undefined });
+      const parsed = parse("a\nb", { columns: undefined } as object);
```
Comment (kt3k) [inline csv/parse_test.ts:979]:
```suggestion
      const parsed = parse("a\nb", {});
```

File: csv/parse_test.ts [hunk 4]
```diff
@@ -994,11 +994,17 @@ Deno.test({
 
     // skipFirstRow option + columns option
     {
-      const parsed = parse("a\nb", { skipFirstRow: false, columns: undefined });
+      const parsed = parse(
+        "a\nb",
+        { skipFirstRow: false, columns: undefined } as { skipFirstRow: false },
```
Comment (kt3k) [inline csv/parse_test.ts:999]:
```suggestion
        { skipFirstRow: false },
```

File: csv/parse_test.ts [hunk 5]
```diff
@@ -994,11 +994,17 @@ Deno.test({
...
+      );
       type _ = AssertTrue<IsExact<typeof parsed, string[][]>>;
     }
     {
-      const parsed = parse("a\nb", { skipFirstRow: true, columns: undefined });
+      const parsed = parse(
+        "a\nb",
+        { skipFirstRow: true, columns: undefined } as { skipFirstRow: true },
```
Comment (kt3k) [inline csv/parse_test.ts:1006]:
```suggestion
        { skipFirstRow: true },
```

File: fmt/bytes_test.ts [hunk 6]
```diff
@@ -87,11 +87,11 @@ Deno.test("format() handles locale option", () => {
...
-  assertEquals(format(-0.4, { locale: undefined }), "-0.4 B");
-  assertEquals(format(0.4, { locale: undefined }), "0.4 B");
-  assertEquals(format(1001, { locale: undefined }), "1 kB");
-  assertEquals(format(10.1, { locale: undefined }), "10.1 B");
-  assertEquals(format(1e30, { locale: undefined }), "1000000 YB");
+  assertEquals(format(-0.4, { locale: undefined } as object), "-0.4 B");
+  assertEquals(format(0.4, { locale: undefined } as object), "0.4 B");
+  assertEquals(format(1001, { locale: undefined } as object), "1 kB");
+  assertEquals(format(10.1, { locale: undefined } as object), "10.1 B");
+  assertEquals(format(1e30, { locale: undefined } as object), "1000000 YB");
```
Comment (kt3k) [inline fmt/bytes_test.ts:94]:
```suggestion
  assertEquals(format(-0.4, {}), "-0.4 B");
  assertEquals(format(0.4, {}), "0.4 B");
  assertEquals(format(1001, {}), "1 kB");
  assertEquals(format(10.1, {}), "10.1 B");
  assertEquals(format(1e30, {}), "1000000 YB");
```

File: fmt/duration_test.ts [hunk 7]
```diff
@@ -85,7 +85,7 @@ Deno.test({
   fn() {
     assertThrows(
       () => {
-        format(16.342, { style: undefined });
+        format(16.342, { style: undefined } as object);
```
Comment (kt3k) [inline fmt/duration_test.ts:88]:
```suggestion
        format(16.342, {});
```

**Comment by kt3k** [comment]
> Though, seeing as tests have to be modified, this seems like a breaking change. 

It seems to me that the changes in test cases are all caused by enabling `exactOptionalPropertyTypes` because there is no changes in public interfaces.

File: fmt/duration_test.ts (hunk 7 above)
Comment (petamoriken) [reply fmt/duration_test.ts:88]: This test is incorrect so I will fix it in another PR.

Comment (petamoriken) [reply fmt/duration_test.ts:88]: #5893

File: fmt/bytes_test.ts (hunk 6 above)
Comment (petamoriken) [reply fmt/bytes_test.ts]: I've placed both `{ locale: undefined }` and `{}` tests.

File: webgpu/texture_with_data.ts [hunk 8]
```diff
@@ -5,14 +5,10 @@ import { describeTextureFormat } from "./describe_texture_format.ts";
...
-    case "1d":
-    case "3d":
-      return 1;
-    case undefined:
-    case "2d":
-      return normalizeExtent3D(texture.size).depthOrArrayLayers ?? 1;
+  if (texture.dimension === "1d" || texture.dimension === "3d") {
+    return 1;
   }
+  return normalizeExtent3D(texture.size).depthOrArrayLayers ?? 1;
```
Comment (petamoriken) [inline webgpu/texture_with_data.ts:11]: It is better to throw an exception when an invalid value is given here.

**Comment by petamoriken** [comment]
Canary CI failed by https://github.com/denoland/deno/pull/25369

> ```
> error: TS2724 [ERROR]: 'Deno' has no exported member named 'ServeTlsOptions'. Did you mean 'ServeOptions'?
>     | Deno.ServeTlsOptions
>            ~~~~~~~~~~~~~~~
>     at file:///home/runner/work/std/std/http/file_server.ts:860:12
> 
> TS2724 [ERROR]: 'Deno' has no exported member named 'ServeTlsOptions'. Did you mean 'ServeOptions'?
>     | (Deno.ServeTlsOptions & Deno.TlsCertifiedKeyOptions) = {
>             ~~~~~~~~~~~~~~~
>     at file:///home/runner/work/std/std/http/file_server.ts:861:13
> 
> Found 2 errors.
> ```

File: http/file_server.ts [hunk 9]
```diff
@@ -855,21 +856,25 @@ function main() {
     console.log(message);
   }
 
+  // TODO(petamoriken): Migrate `Deno.ServeTcpOptions | (Deno.ServeTcpOptions & Deno.TlsCertifiedKeyOptions)` in v2
```
Comment (kt3k) [inline http/file_server.ts:859]: What does this mean?

Comment (petamoriken) [reply http/file_server.ts:859]: I used `Deno.ServeTlsOptions`, but it was renamed to `Deno.ServeTcpOptions` in Deno v2 (https://github.com/denoland/deno/pull/25369), so I wrote flatly this type 😭

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**